src/dmd/root/filename.d

```diff
@@ -49,30 +50,25 @@ nothrow:
         this.str = mem.xstrdup(str);
     }
 
-    extern (C++) bool equals(const RootObject obj) const pure
-    {
-        return compare(obj) == 0;
-    }
-
+    /// Compare two name according to the platform's rules (case sensitive or not)
     extern (C++) static bool equals(const(char)* name1, const(char)* name2) pure
     {
-        return compare(name1, name2) == 0;
+        return equals(name1.toDString, name2.toDString);
     }
 
-    extern (C++) int compare(const RootObject obj) const pure
+    /// Ditto
+    extern (D) static bool equals(const(char)[] name1, const(char)[] name2) pure
     {
-        return compare(str, (cast(FileName*)obj).str);
-    }
+        if (name1.length != name2.length)
+            return false;
 
-    extern (C++) static int compare(const(char)* name1, const(char)* name2) pure
-    {
         version (Windows)
         {
-            return stricmp(name1, name2);
+            return strnicmp(name1.ptr, name2.ptr, name1.length) == 0;
```

Comment:
> These POSIX functions are deprecated. Use the ISO C++ conformant _strnicmp, _wcsnicmp, _mbsnicmp, _strnicmp_l, _wcsnicmp_l, _mbsnicmp_l instead.

https://msdn.microsoft.com/en-us/library/ms235324.aspx

Reply:
I see strncmp has reappeared in another guise. Please don't use it. I pointed this out in a previous PR, and why.

Reply:
Previous review of strncmp:
https://github.com/dlang/dmd/pull/8575#pullrequestreview-147490646
"strncmp is the wrong answer here. I quote the C Standard: "The strncmp function compares not more than n characters (characters that follow a null character are not compared) from the array pointed to by s1 to the array pointed to bys 2." I.e. embedded 0's are still a factor. Use memcmp() and memicmp() instead. I've never found a legitimate use for strncmp()."